### ISSUE TITLE
Syncthing deletes files after file renamed

### ISSUE THREAD

**Issue by RaysCars** [syncthing/syncthing#9890]
### What happened?

When files are renamed in a folder monitored by syncthing, syncthing deletes the files. The folder has staggered file versioning which allows me to recover. This occurs for WIndows version and Linux version.
Windows verion: 
syncthing v1.28.1 "Gold Grasshopper" (go1.23.3 windows-amd64) builder@github.syncthing.net
Linux version:
syncthing v1.28.1 "Gold Grasshopper" (go1.23.3 linux-arm64) docker@github.syncthing.net

### Syncthing version

v1.28.1

### Platform & operating system

Windows 11

### Browser version

firefox 133.03

**Comment by calmh** [comment]
Can you be more detailed about how to reproduce this?

**Comment by RaysCars** [comment]
Not much more to describe. Renaming a file or group of files causes sycthing to delete the files on the next refresh cycle. It's not only this version, I've noticed tihs effect over the past year. Whether I rename files from Windows explorer or from pwsh /cmd prompt, I'll see the sync occur on the target side then files delete on both sides.

**Comment by rasa** [comment]
Not to discount your experience, but I have never seen this occur, and have been using Syncthing for years on Windows, Linux, Android, and even iOS. My guess is there is something unique to your setup. Here's an incomplete list of things to consider, from likely to not:
1. Syncthing index file corruption (try rebuilding them)
2. Running on a non-standard filesystem: Symlinks, directory junctions, or dedup files/directories, NAS, samba, network shares, etc. (try sharing from a physical drive)
3. You've caught a virus or malware (try running a deep scan)
4. Some Anti-virus/anti-malware software is running (check their logs, or temporarily disable)
5. A pseudo-filesystem software is also running (GoogleDrive, ShareFile, etc.) (perhaps boot in safe mode to disable)
6. Hard disk corruption (check your system's logs)
7. Bad RAM (run a memory checker)
8. Something else?

If everything checks out, we'll need more information on your setup to help you solve this (logs, config, steps, etc.)

**Comment by RaysCars** [comment]
I think everything checks out.

**Comment by imsodin** [comment]
As already explained, with the limited, general information you provided, that scenario is definitely not happening widely. To fix your problem, we need to know what exactly is happening.Please provide one of two things (or both of course):

- Steps to reproduce from scratch  
A list of simple steps anyone can follow. First specify the systems (OS, filesystem, ...) in use. And the data to sync must be created as part of the steps. If the starting point is "your system" including existing state (data, database, ...), that unfortunately isn't useful.
- Logs while you reproduce in your system  
To give you more targeted instructions which logs exactly, please trigger the problem in your system again and describe in as much detail as possible what you do and what you observe. Then we can either ask relevant followup questions or tell you what debug logging to enable to capture logs while you trigger it again.

**Comment by RaysCars** [comment]
when I rename 1 or more files on Windows or Linux folders, the files delete. I haven't tried removing file versionong.
using powershell: 
`dir 2024_12* -file | %{ $n = $_.name -replace "_","-" ; Rename-Item $_ $n }`

using Windows explorer: right click 1 or multiple files, Rename
![image](https://github.com/user-attachments/assets/8e11f5fb-562f-4ff5-98a9-2f8e3a01dcd9)


It's as simple as that.

**Comment by rasa** [comment]
@RaysCars Again, I have never seen this occur, and I've been using Syncthing on Windows 7, 10, and now 11 for years. You provided the steps that lead to the issue, but as explained in @imsodin's [comment](https://github.com/syncthing/syncthing/issues/9890#issuecomment-2564325245), we'll need more information to figure out what's going on. At a minimum, we'll need the specific OS version, underlying filesystem (NTFS, FAT, etc), config settings, relevant logs, etc.

**Comment by uok** [comment]
Do you mean Syncthing deletes files on same device? E.g. you rename on device A and Syncthing then deletes files on device A because device B does not know them?

**Comment by DenVladi** [comment]
> Можете ли вы подробнее рассказать, как это воспроизвести?

Создаём на компьютере 3 файла, "file 1.txt", "file-1.txt", "file_1.txt". Синхронизируемся с телефоном. Если удалить файл "file_1.txt" или переименовать в "file.1.txt" то после синхронизации удаляются все 3 файла.

**Comment by RaysCars** [comment]
This same behaviour occurs with me.

**Comment by calmh** [comment]
Translated from russian,

> Create 3 files on your computer, "file 1.txt", "file-1.txt", "file_1.txt". Synchronize with your phone. If you delete the file "file_1.txt" or rename it to "file.1.txt", then all 3 files are deleted after synchronization.

This is a classic old Android media library bug, where it does not differentiate between space, dash or underscore and will delete one when another is renamed. We used to have it way back when interacting with that media framework, and fixed it by not doing that any more.

In any case, for this to be a Syncthing bug there needs to not be a phone involved...

**Comment by DenVladi** [comment]
> Это классический баг старой медиабиблиотеки Android, где он не различает пробел, тире или подчеркивание и удаляет один при переименовании другого. Мы использовали его давно при взаимодействии с этим медиафреймворком и исправили его, больше так не делая.
> 
> В любом случае, чтобы это была ошибка Syncthing, телефон не должен быть замешан...

Android 13 / UI 5.1
Пробовал удалять на телефоне эти файлы, всё хорошо. 
Проблема происходит когда удаляешь с компьютера файл с подчёркиванием "file_1.txt".

**Comment by calmh** [comment]
In English, here.

**Comment by calmh** [comment]
Yeah, again, this is _not Syncthing_. Just the fact that it's dependent on what the file name is should be a strong indicator to everyone that something else is going on.
